src/rust/ide/view/src/documentation.rs

```diff
@@ -68,6 +78,7 @@ pub struct ViewModel {
     /// to EnsoGL shapes, and pass them to the DOM instead.
     overlay        : component::ShapeView<overlay::Shape>,
     display_object : display::object::Instance,
+    closures       : Rc<CloneCell<Vec<OptClosure>>>
```

Comment:
1. "closures" is like naming a variable "var"
2. `CloneCell<Vec<...>>` has a very high performance cost. It is clonning the whole vector on each access. I don't believe you want this.

Reply:
> 2. `CloneCell<Vec<...>>` has a very high performance cost. It is clonning the whole vector on each access. I don't believe you want this.

Yes, we want. Because we never want to read this vector, only to set.

Reply:
> Why the `ViewModel` needs to be `CloneRef`? I did not have problems with that earlier, but now, if we clone ref it, we can easily leak accidenatally the closures, so I'm wondering why it needs to be clonable at all.

The Clone-refable model used in FRP was one of the most common patterns in GUI. And yes, it caused a lot of problems, (for example not-disappearing nodes on the graph scene). But we need clone-refable structures in current FRP implementation

Reply:
> Could you also explain e the live-time of the ViewModel? When it is created, who owns it and its copies, and when it will be deleted, so I could better understand how long the closures will live here? Also, add this info to the docs.

TBH I'm surprised with this requirement. NONE of our views' `Model`s have such detailed description and they may leak as well. Are we going to extend docs in all of the models?

Reply:
> The Clone-refable model used in FRP was one of the most common patterns in GUI. And yes, it caused a lot of problems, (for example not-disappearing nodes on the graph scene)

I know you made a big mental shortcut here (based on our phone call just a minute ago), so to explain to readers, clone-refable models never caused any problems in GUI. What caused the problem was clone-refable FRP networks, which we are slowly improving. Clone-refable models are rather safe, but they can be made even safer if we would allow FRP networks to manage their lifetimes in the future.

> But we need clone-refable structures in current FRP implementation
In other places we are using just the name `Model`, not `ViewModel`, this made confusion in my head what kind of model is this. So to keep it consistent, @BinarySoftware rename it to `Model`, please

> TBH I'm surprised with this requirement. 
100% agree. If these are just normal FRP models, I'm ok with it. The only part that brought confusion was the `Viewodel` name here.